vm/src/stdlib/array.rs

```diff
@@ -99,17 +99,17 @@ macro_rules! def_array_enum {
             fn count(&self, obj: PyObjectRef, vm: &VirtualMachine) -> PyResult<usize> {
                 match self {
                     $(ArrayContentType::$n(v) => {
-                        let val = $t::try_from_object(vm, obj)?;
-                        Ok(v.iter().filter(|&&a| a == val).count())
+                        let val = <Option<$t>>::try_from_object(vm, obj)?;
+                        Ok(v.iter().filter(|&&a| Some(a) == val).count())
```

Comment:
I think this is the correct way to get the value of `val`.
Because any value in array cannot be `None`, I think the actual comparison is not required.
```suggestion
                        Ok(<Option<$t>>::try_from_object(vm, obj)?.map_or(0, |val| {
                            v.iter().filter(|&&a| a == val).count()
                        })
```

Reply:
Thank you for your advise 👍 I edit my commit :)